src/nodes.coffee

```diff
@@ -3338,6 +3353,48 @@ exports.Assign = class Assign extends Base
   unfoldSoak: (o) ->
     unfoldSoak o, this, 'variable'
 
+  addScopeVariables: (o, {checkAssignability = yes} = {}) ->
+    return unless not @context or @context is '**='
+
+    varBase = @variable.unwrapAll()
+    if checkAssignability and not varBase.isAssignable()
+      @variable.error "'#{@variable.compile o}' can't be assigned"
+
+    varBase.eachName (name) =>
+      return if name.hasProperties?()
+
+      message = isUnassignable name.value
+      name.error message if message
+
+      # `moduleDeclaration` can be `'import'` or `'export'`.
+      @checkNameAssignability o, name
+      if @moduleDeclaration
+        o.scope.add name.value, @moduleDeclaration
+        name.isDeclaration = yes
+      else if @param
+        o.scope.add name.value,
+          if @param is 'alwaysDeclare'
+            'var'
+          else
+            'param'
+      else
+        alreadyDeclared = o.scope.find name.value
+        name.isDeclaration ?= not alreadyDeclared
```

Comment:
The typical case, the left hand of an assignment represents a declaration unless it already exists